File: src/main/java/de/flavormate/ba_entities/story/repository/StoryRepository.java
/* Licensed under AGPLv3 2024 */
package de.flavormate.ba_entities.story.repository;

import de.flavormate.ba_entities.story.model.Story;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface StoryRepository extends JpaRepository<Story, Long> {

  void deleteAllByRecipeId(Long id);

  @Query("select s from Story s where lower(s.label) like lower(concat('%', ?1, '%'))")
  Page<Story> findBySearch(String searchTerm, Pageable pageable);
}

File: src/main/kotlin/de/flavormate/features/story/repositories/StoryRepository.kt
/* Licensed under AGPLv3 2024 - 2025 */
package de.flavormate.features.story.repositories

import de.flavormate.features.story.daos.models.StoryEntity
import io.quarkus.hibernate.orm.panache.kotlin.PanacheQuery
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepositoryBase
import io.quarkus.panache.common.Sort
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class StoryRepository : PanacheRepositoryBase<StoryEntity, String> {
  override fun findAll(sort: Sort): PanacheQuery<StoryEntity> {
    return find(query = "select s from StoryEntity s", sort = sort)
  }

  fun findBySearch(query: String, sort: Sort): PanacheQuery<StoryEntity> {
    val params = mapOf("query" to query)

    return find(
      query =
        "select s from StoryEntity s where lower(s.label) like lower(concat('%', :query, '%'))",
      sort = sort,
      params = params,
    )
  }
}
